code/game/gamemodes/traitor/traitor.dm

```diff
@@ -52,13 +48,67 @@
 		return FALSE
 	return TRUE
 
-
 /datum/game_mode/traitor/post_setup()
-	for(var/t in pre_traitors)
-		var/datum/mind/traitor = t
-		traitor.add_antag_datum(/datum/antagonist/traitor)
-	..()
+	. = ..()
+
+	for(var/datum/mind/traitor in pre_traitors)
+		var/datum/antagonist/traitor/traitor_datum = new(src)
+		if(isAI(traitor.current))
+			traitor.add_antag_datum(traitor_datum)
+			continue
 
+		traitor_datum.delayed_objectives = TRUE
+		traitor.add_antag_datum(traitor_datum)
+
+	if(length(pre_traitors))
+		var/random_time = rand(300 SECONDS, 900 SECONDS)
```

Comment:
```suggestion
		var/random_time = rand(5 MINUTES, 15 MINUTES)
```

Reply:
Ah I pushed that change I thought, good catch!